src/EditorFeatures/Core/InheritanceMargin/InheritanceMarginServiceHelpers.cs

```diff
@@ -138,20 +145,36 @@ internal static class InheritanceMarginServiceHelper
             // Go down the inheritance chain to find all the overrides targets.
             var allOverriddenSymbols = await SymbolFinder.FindOverridesArrayAsync(memberSymbol, solution, cancellationToken: cancellationToken).ConfigureAwait(false);
 
-            // Go up the inheritance chain to find all the implemented targets.
-            var implementingSymbols = memberSymbol.ExplicitOrImplicitInterfaceImplementations();
-
             // Go up the inheritance chain to find all overriding targets
             var overridingSymbols = GetOverridingSymbols(memberSymbol);
 
+            // Go up the inheritance chain to find all the implemented targets.
+            var allImplementingSymbols = GetImplementingSymbolsForTypeMember(memberSymbol, overridingSymbols);
+
             // Go down the inheritance chain to find all the implementing targets.
-            var allImplementedSymbols = await GetImplementedSymbolsAsync(solution, memberSymbol, cancellationToken).ConfigureAwait(false);
+            var allImplementedSymbols = await GetImplementedSymbolsForTypeMemberAsync(solution, memberSymbol, cancellationToken).ConfigureAwait(false);
```

Comment:
this doesn't seem to be using Distinct/OriginalDefinition either.

Reply:
All the calls are moved to the top helpers, : ) mark this as resolved.